Content.Server/_NF/Storage/AnchorableStorageSystem.cs

```diff
@@ -0,0 +1,115 @@
+using System.Linq;
+using Content.Server.Popups;
+using Content.Shared.Construction.Components;
+using Content.Shared.Mind.Components;
+using Content.Shared.Nyanotrasen.Item.PseudoItem;
+using Content.Shared.Storage;
+using JetBrains.Annotations;
+using Robust.Server.GameObjects;
+using Robust.Shared.Containers;
+using Robust.Shared.Map.Components;
+
+namespace Content.Server._NF.Storage;
+
+/// <summary>
+/// This is used for restricting anchor operations on storage (one bag max per tile)
+/// and ejecting living contents on anchor.
+/// </summary>
+public sealed class AnchorableStorageSystem : EntitySystem
+{
+    [Dependency] private readonly MapSystem _map = default!;
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly TransformSystem _xform = default!;
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<AnchorableStorageComponent, AnchorStateChangedEvent>(OnAnchorStateChanged);
+        SubscribeLocalEvent<AnchorableStorageComponent, AnchorAttemptEvent>(OnAnchorAttempt);
+        SubscribeLocalEvent<AnchorableStorageComponent, ContainerIsInsertingAttemptEvent>(OnInsertAttempt);
+    }
+
+    private void OnAnchorStateChanged(Entity<AnchorableStorageComponent> ent, ref AnchorStateChangedEvent args)
+    {
+        if (!args.Anchored)
+            return;
+
+        if (CheckOverlap((ent, ent.Comp, Transform(ent))))
+        {
+            _popup.PopupEntity(Loc.GetString("anchored-storage-already-present"), ent);
+            _xform.Unanchor(ent, Transform(ent));
+            return;
+        }
+
+        // Eject any sapient creatures inside the storage.
+        // Does not recurse down into bags in bags - player characters are the largest concern, and they'll only fit in duffelbags.
+        if (!TryComp(ent.Owner, out StorageComponent? storage))
+            return;
+
+        var entsToRemove = storage.StoredItems.Keys.Where(storedItem =>
+                HasComp<MindContainerComponent>(storedItem)
+                || HasComp<PseudoItemComponent>(storedItem)
+            ).ToList();
```

Comment:
😭 
```suggestion
        var entsToRemove = storage.StoredItems.Keys.Where(storedItem =>
                HasComp<MindContainerComponent>(storedItem)
                || HasComp<PseudoItemComponent>(storedItem))
            .ToList();
```

Reply:
Frankly, I like the parenthesis giving something that the method is working on rather than a hanging function performed on the ether.

Reply:
I think breaking LINQ queries using dot-syntax is generally preferred but yeah that's also understandable